python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -1202,6 +1202,52 @@ def test_packing(self):
         assert to_launch == {}
         assert not rem
 
+    def test_skip_placed_bundles(self):
+        # test that we do not launch new nodes for bundles that are already placed.
+        provider = MockProvider()
+        scheduler = ResourceDemandScheduler(
+            provider,
+            TYPES_A,
+            10,
+            head_node_type="p2.8xlarge",
+            upscaling_speed=1,
+        )
+
+        provider.create_node({}, {TAG_RAY_USER_NODE_TYPE: "p2.8xlarge"}, 1)
+        # At this point our cluster has 1 p2.8xlarge instances (8 GPUs) and is
+        # fully idle.
+        nodes = provider.non_terminated_nodes({})
```

Comment:
Does this simulate the case that would happen at runtime (`node_1` doesn't exist in `nodes`)?

Reply:
I replaced the node_1 with the existing node in the new commit.